asdf/block.py

```diff
@@ -1005,6 +1009,9 @@ def read(self, fd, past_magic=False, validate_checksum=False):
             If `True`, validate the data against the checksum, and
             raise a `ValueError` if the data doesn't match.
         """
+        if self._data_callback is not None:
```

Comment:
Shouldn't the docstring for this method also point this out?

Reply:
Thanks! I added some descriptions to the docstring. Let me know if this looks good to you.